src/Service/Configurations/RuntimeConfigValidator.cs

```diff
@@ -229,25 +232,96 @@ public static void ValidateEntitiesDoNotGenerateDuplicateQueriesOrMutation(Runti
+                    {
+                        pathForEntity = entity.Rest.Path is not null ? entity.Rest.Path.TrimStart('/') : entityName;
+                        isRestEnabledForEntity = entity.Rest.Enabled;
+                        SupportedHttpVerb[] supportedOperations = entity.Rest.Methods;
+
+                        // Since 'path' is an optional property, we skip validation if its absent.
+                        ValidateAndGetRestSettingsForEntity(entityName, pathForEntity);
+                    }
+
+                    if (isRestEnabledForEntity && !restPathsForEntities.Add(pathForEntity))
+                    {
+                        // Presence of multiple entities having the same rest path configured causes conflict.
+                        throw new DataApiBuilderException(
+                            message: $"The rest path: {pathForEntity} specified for entity: {entityName} is already used by another entity.",
+                            statusCode: HttpStatusCode.ServiceUnavailable,
+                            subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError
+                            );
+                    }
+                }
+
+                // If GraphQL endpoint is disabled globally, we skip the validations related to it.
+                if (!runtimeConfig.Runtime.GraphQL.Enabled)
+                {
+                    continue;
+                }
+
+                if (entity.GraphQL is null)
+                {
+                    continue;
+                }
+                else if (entity.GraphQL.Enabled)
                 {
                     ValidateNameRequirements(entity.GraphQL.Singular);
                     ValidateNameRequirements(entity.GraphQL.Plural);
                 }
             }
         }
 
+        /// <summary>
+        /// Helper method to get the rest path/rest enabled properties for the entity if it is correctly configured.
+        /// The rest path should not be null/empty and should not conflict with the rest path
+        /// configured for any other entity.
+        /// </summary>
+        /// <param name="entityName">Name of the entity.</param>
+        /// <param name="pathForEntity">The rest path for the entity.</param>
+        /// <exception cref="DataApiBuilderException">Throws exception when rest path contains an unexpected value.</exception>
+        private static void ValidateAndGetRestSettingsForEntity(string entityName, string pathForEntity)
+        {
+            if (string.IsNullOrEmpty(pathForEntity))
+            {
+                // The rest 'path' cannot be empty.
+                throw new DataApiBuilderException(
+                    message: $"The rest path for entity: {entityName} cannot be empty.",
+                    statusCode: HttpStatusCode.ServiceUnavailable,
+                    subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError
+                    );
```

Comment:
Would this ever be triggered when the user explicitly does not include a rest path? e.g. they don't want to customize the rest path.

Reply:
So the rest path will always be there - either it would be the top level entity name or the custom rest path. The validation will be performed in either cases.